packages/openapi-v3/src/json-to-schema.ts

```diff
@@ -75,6 +75,13 @@ export function jsonToSchemaObject(jsonDef: JsonDefinition): SchemaObject {
 
         break;
       }
+      case '$ref': {
+        result.$ref = json.$ref.replace(
+          '#/definitions',
+          '#/components/schemas',
```

Comment:
Where is the original value containing `#/definitions` coming from? Can we fix the producer of `$ref` values instead of patching references here? I am ok to defer that change until our entire REST stack is switched over to OAIv3, as long as we keep track of this task.

Reply:
@bajtos in a JSON Schema, a property's `$ref` has to be `#/definitions`, e.g.:

```ts
// A model's JSON definition:
{
  properties: {
    id: {type: 'string'},
    child: {$ref: '#/definitions/child'}
  }
}
```

Convert to OpenAPI schema:

```ts
// A model's OpenAPI schema:
{
  properties: {
    id: {type: 'string'},
    child: {$ref: '#/components/schemas/child'}
  }
}
```
So I believe the conversion has to be here.

And this is the corresponding [test case](https://github.com/strongloop/loopback-next/pull/1046/commits/0b47d3efedce3a62822d4868b6e90e74f980658d#diff-777a8b2219237f64729cbc32192cb356R58)

Does it answer your question? If you still have concern I will create an issue for it.